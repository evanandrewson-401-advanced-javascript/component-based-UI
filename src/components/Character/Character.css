.Character {
    margin: 1em;
    position: relative;    
    h1 {
        position: absolute;
        top: 230px;
        height: 70px;
        width: 300px;
        background-color: black;
        opacity: .6;
        margin: 0;
        color: white;
        box-sizing: border-box;
        display: flex;
        justify-content: center;
        align-items: center;
    }
    .species {
        display: flex;
        width: 300px;
        background-color:black;
        opacity: .6;
        justify-content: space-between;
        margin: 0;
        padding: 5px;
        box-sizing: border-box;
        font-size: 1.5rem;

        #left {
            color: white;
        }

        #right {
            color: orange;
        }
    }
    img {
        margin: 0;
    }
}