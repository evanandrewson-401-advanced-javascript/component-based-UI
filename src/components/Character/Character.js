import React from 'react';
import PropTypes from 'prop-types';

const Character = (image, name, species) => (
  <>
    <img src={image} />
    <h1>{name}</h1>
    <p>{species}</p>
  </>
);

Character.propTypes = {
  image: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired
}

export default Character;