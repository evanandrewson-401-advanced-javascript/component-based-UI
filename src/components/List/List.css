.List {
    display: flex;
    flex-wrap: wrap;
    background-color: #202329;
    margin: 0;
    
    li {
        list-style: none;
    }
}