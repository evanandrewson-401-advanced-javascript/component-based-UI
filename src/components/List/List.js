import React from 'react';
import Character from '../Character/Character';
import PropTypes from 'prop-types';
import styles from './List.css';

const List = ({ itemsArray, Component }) => (
  <ul className={styles.List}>
    {itemsArray.map(item => 
      (<li key={item.name}>
        <Component {...item}/>
      </li>))}
  </ul>);

List.propTypes = {
  itemsArray: PropTypes.array.isRequired,
  Component: PropTypes.func.isRequired
}


export default List;