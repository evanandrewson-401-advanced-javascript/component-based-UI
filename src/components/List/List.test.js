import React from 'react';
import { shallow } from 'enzyme';
import List from './List.js';
import characters from '../../data';
import Character from '../Character/Character'

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(<List 
      itemsArray={characters}
      Component={Character}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});