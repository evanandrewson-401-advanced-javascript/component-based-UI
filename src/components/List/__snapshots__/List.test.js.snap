// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ul
  className="List"
>
  <li
    key="Rick Sanchez"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      name="Rick Sanchez"
      species="Human"
    />
  </li>
  <li
    key="Morty Smith"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      name="Morty Smith"
      species="Human"
    />
  </li>
  <li
    key="Summer Smith"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      name="Summer Smith"
      species="Human"
    />
  </li>
  <li
    key="Beth Smith"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      name="Beth Smith"
      species="Human"
    />
  </li>
  <li
    key="Jerry Smith"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      name="Jerry Smith"
      species="Human"
    />
  </li>
  <li
    key="Abadango Cluster Princess"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      name="Abadango Cluster Princess"
      species="Alien"
    />
  </li>
  <li
    key="Abradolf Lincler"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      name="Abradolf Lincler"
      species="Human"
    />
  </li>
  <li
    key="Adjudicator Rick"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      name="Adjudicator Rick"
      species="Human"
    />
  </li>
  <li
    key="Agency Director"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      name="Agency Director"
      species="Human"
    />
  </li>
  <li
    key="Alan Rails"
  >
    <Character
      image="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      name="Alan Rails"
      species="Human"
    />
  </li>
</ul>
`;
